code/modules/vending/_vending.dm

```diff
@@ -208,6 +208,12 @@
 	. = ..()
 	wires = new /datum/wires/vending(src)
 
+	if(SStts.tts_enabled)
+		var/static/vendor_voice_by_type = list()
+		if(!vendor_voice_by_type[type])
+			vendor_voice_by_type[type] = pick(SStts.available_speakers)
+		voice = vendor_voice_by_type[type]
```

Comment:
and here. does this just not support configs changing? why does this need to be locked behind the preference?

Reply:
yes, because again it's linking up with a http server, the voices need to be cached beforehand. The config needs to be set before SStts init or else it doesn't load at all.